src/python/BuildingControlsSimulator/BuildingModels/EnergyPlusBuildingModel.py

```diff
@@ -32,142 +51,359 @@ class EnergyPlusBuildingModel(BuildingModel):
-        """Gets occupied zones from zones that have a tstat in them."""
-        return [
-            tstat.Zone_or_ZoneList_Name
-            for tstat in self.idf.ep_idf.idfobjects[
-                "zonecontrol:thermostat".upper()
-            ]
-        ]
+        proc = subprocess.run(shlex.split(cmd), stdout=subprocess.PIPE)
+        if not proc.stdout:
+            raise ValueError(
+                f"Empty STDOUT. Invalid EnergyPlusToFMU cmd={cmd}"
+            )
+
+        # EnergyPlusToFMU puts fmu in cwd always, move out of cwd
+        shutil.move(
+            os.path.join(os.getcwd(), self.init_fmu_name), self.fmu_path,
+        )
+        # check FMI compliance
+        # -h specifies the step size in seconds, -s is the stop time in seconds.
+        # Stop time must be a multiple of 86400.
+        # The step size needs to be the same as the .idf file specifies
+
+        cmd = (
+            "yes |"
+            f" {self.ext_dir}/FMUComplianceChecker/fmuCheck.linux64"
+            f" -h {self.timesteps_per_hour}"
+            " -s 172800"
+            f" {self.fmu_path}"
+        )
+        # subprocess.run(cmd.split(), stdout=subprocess.PIPE)
+        # if not proc.stdout:
+        #     raise ValueError(f"Empty STDOUT. Invalid EnergyPlusToFMU cmd={cmd}")
+
+        return self.fmu_path
 
-    def actuate_HVAC_equipment_init_fmu(self, step_HVAC_mode, init_fmu):
+    def initialize(self, t_start, t_end, t_step, categories_dict={}):
         """
         """
-        if self.cur_HVAC_mode != step_HVAC_mode:
-            if step_HVAC_mode == HVAC_modes.SINGLE_HEATING_SETPOINT:
-                init_fmu.set(
-                    self.idf.FMU_control_type_name, int(step_HVAC_mode)
-                )
-                init_fmu.set(
-                    self.idf.FMU_control_heating_stp_name, self.T_heat_on
-                )
-                init_fmu.set(
-                    self.idf.FMU_control_cooling_stp_name, self.T_cool_off
-                )
-            elif step_HVAC_mode == HVAC_modes.SINGLE_COOLING_SETPOINT:
-                init_fmu.set(
-                    self.idf.FMU_control_type_name, int(step_HVAC_mode)
-                )
-                init_fmu.set(
-                    self.idf.FMU_control_heating_stp_name, self.T_heat_off
-                )
-                init_fmu.set(
-                    self.idf.FMU_control_cooling_stp_name, self.T_cool_on
-                )
+        logger.info(f"Initializing EnergyPlusBuildingModel: {self.fmu_path}")
+        self.allocate_output_memory(t_start, t_end, t_step, categories_dict)
+        self.init_step_output()
 
-            elif step_HVAC_mode == HVAC_modes.UNCONTROLLED:
-                init_fmu.set(
-                    self.idf.FMU_control_type_name, int(step_HVAC_mode)
+        self.fmu = pyfmi.load_fmu(fmu=self.fmu_path)
+        self.fmu.initialize(t_start, t_end)
+
+    def tear_down(self):
+        """tear down FMU"""
+        # self.fmu.terminate()
```

Comment:
dead code?

Reply:
yupe, im removing this and adding a note about the model cleanup.